Cybersecurity_Tools/PyRecon/pyrecon/plugins/tls_parser.py

```
@@ -0,0 +1,414 @@
+        Returns:
+            List of certificates in chain
+        """
+        chain = []
+        
+        try:
+            context = ssl.create_default_context()
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_NONE
+            
+            with socket.create_connection((host, port), timeout=self.timeout) as sock:
+                with context.wrap_socket(sock, server_hostname=host) as ssock:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)